### ISSUE TITLE
Turning on expansions by clicking its icon in SE

### ISSUE THREAD

**Issue by chrismaliszewski** [NPBruce/valkyrie#1072]
I have an idea for the future release (not now) which would make turning expansions easier and faster for a user.

The idea is, clicking an expansion's icon either:
a. moves a user directly to the Select Content's submenu with the expansion, where he/she can turn it on or off; clicking Back button causes a return to the SE (requires knowing what was a previous user's location in the app) to the same position (no reload of the list! Just a reload of the scenario's line).
or
b. toggles the expansion's activation; it'd require refreshing the scenario's line (not the whole list).

**Comment by BenRQ** [comment]
I'm not sure this enhancement is really required, this sounds like a very small improvement.
Most users would just go to select content.

And for B, I think some people would just press expansions icons by mistake while scrolling, and they won't understand what's happening with the refresh, and why they have the content activated.
And with the size of the expansions icons, this would not be usable on any mobile device.

I would put this one at the end of the todo list.
Please share your thoughts if you think otherwise.

**Comment by chrismaliszewski** [comment]
Of course. It isn't something major. 

Below I write my thoughts about the whole idea for the future reference.

Now knowing the app I can only imagine what it's to be a fresh user. But let's try:
1. Let's say, I have a physical base game and two expansions. 
2. I run the app and click New game because I don't know what 'Select Content' does and I am interested in starting a new game immediately.
3. I want to play a game which uses all of my expansions so I look for a scenario which all of the icons associated with them (here the hover effect https://github.com/NPBruce/valkyrie/issues/1073 would be handy if I don't know the icons). 
4. I find a suitable scenario but I notice that I am missing some expansions.
5. At this point, I need to think "how can I turn them on?". What I would do, would be clicking/tapping the expansion's name. If that doesn't work, I'd go back and check other menu options to find the answer and eventually I'd click 'Select Content'. 

So, yes, it'd be a small improvement but the one that could make it easier and faster to turn on missing expansions. 

**In terms of usability.** 
I wouldn't be afraid of the problem of clicking icons by mistake if they are displayed clearly as buttons instead of text. Also activating/deactivating possibility to play a scenario should be clear after from one to two clicks. I hope people are that intelligent :).

**I'd be rather worried about two things:**
1. Strange behaviour of UI when clicking the buttons which I think could be hooked under your statement 'they won't understand what's happening with the refresh'. After a user clicks an icon it "magically" disappear and appear somewhere else in the line. It isn't something common, might be unexpected and confusing.
2. As you mentioned, the size of the buttons. Not even just for mobile devices but all. First of all, as mentioned above, the icons should be buttons. Second, they should be more separated than they are now (fortunately, there is enough space in terms of width for that). But it still leaves the problem of the height of the buttons as there isn't enough space as for now. A scenario line could be a little larger but it'd require testing.

**Comment by BenRQ** [comment]
Regarding the height (vertical size) of each scenario line, the purpose of this first version of scenario explorer was to keep the line as small as possible, in order to show as many scenarios as possible, while it stays practical and easy to read.

Making this bigger would allow more flexibility for improvement, but less scenarios on screen.

**Comment by chrismaliszewski** [comment]
So maybe something else.

The longest scenario's name so far is the follwing:
![image](https://user-images.githubusercontent.com/7532602/53083408-c2ef2e00-3563-11e9-9617-db3b740a20cf.png)
About 46 characters. 

And the scenarios with the numerous expansions are:
![image](https://user-images.githubusercontent.com/7532602/53083534-ffbb2500-3563-11e9-89e4-5e28031629d8.png)
4 in total. 

If the number of expansions was all 5 expansions with a little bit smaller font (~2 points smaller) a scenario line could look something like that:
![image](https://user-images.githubusercontent.com/7532602/53084730-4f025500-3566-11e9-840e-899569348273.png)
1. The 'Missing Expansions' text could be removed and instead the previous version of the red/black icons could be used. But this time the expansions' icons should look like buttons/something clickable.
2. The design wouldn't require refreshing the whole line but just the icon's color and the play/stop button. 
3. If the red icons are not clear enough that there is something wrong and that's the reson why there is the 'Stop' button there, a hovering effect on the 'Stop' button could be created (https://github.com/NPBruce/valkyrie/issues/1073) with an explanation why the scenario is not playable. 
4. It'd be possible to earn more vertical space and put even more scenarios on the screen :).

**Comment by chrismaliszewski** [comment]
I have an important question to the topic. I've just opened Editor to check how many expansions exist and I was surprised seeing this list:
![image](https://user-images.githubusercontent.com/7532602/53085083-0008ef80-3567-11e9-87a9-f3467a30f482.png)
Why so many of them have the same icon? If they are marked as 'the same', would I be able to play any scenario with a moon icon after buying just one expansion of the kind? If yes, why the list of scenario is not narrower if it doesn't matter? If no, why they are marked the same (or almost the same)?

**Comment by BenRQ** [comment]
You did not take into account the number of flags.
I moved the expansions below the title to leave some space for the flags.

Regarding the expansions, it is just the way it has been released by FFG I guess.
It is up to the scenario author to decide if the player should have the full expansion or just part of it (like just the tiles, or just the monster for example).

**Comment by chrismaliszewski** [comment]
You're right again. And I had a good example right above :p. 
To that issue there a simple solution:
1. Limit a title to 50 characters. 
2. Create the languages part in a way the languages' icons wrap is necessary as presented in the below image.
![image](https://user-images.githubusercontent.com/7532602/53086381-ddc4a100-3569-11e9-9c86-9cad00489222.png)
But there is still this issue of possible dozen of expansions' icons... If there were just 5, it'd be a good design but no one can guarantee that :|.

**Comment by chrismaliszewski** [comment]
Coming back to the expansions list. From what I understand, MoM1EI has a different set of tiles/monsters/etc than others with the key icon, right? 
So, could you elaborate:
1. Why are they marked the same?
2. How would a user know which expansion he/she is missing if the only thing he/she sees is the icon?
3. I've just checked with a local scenario. I set Required Expansions to MoM1ET. After that, I activated RN expansion in the 'Select Content'. Why the scenario is active (I can play it) although the scenario required a different expansion and the only thing that both have in common is the icon? Aren't they different thing with different items?

If everything is fine, shouldn't the list of icons be limited to just these 5 icons instead of having something like that 
![image](https://user-images.githubusercontent.com/7532602/53087084-38123180-356b-11e9-9d7a-0a5dc4d2eae5.png)
two of the same icons because someone selected the expansion with a candle icon and SM expansion? It's a little messy here.

Of course, it's a topic for another big ticket but could we discuss it shortly here for the issue's reference?

**Comment by BenRQ** [comment]
RN contains MOM1ET MOM1EM MOM1EI
SM contains CotWT CotWM CotWI FAI FAM FAT

![image](https://user-images.githubusercontent.com/40794717/53087829-5bf05b80-3508-11e9-856e-05fda1e1520e.png)

**Comment by BenRQ** [comment]
> two of the same icons in the single scenario because someone selected the expansion with a candle icon and SM expansion? It's a little messy here.

Yes if you select SM as a required expansions in your scenario, you don't need to select the others.

**Comment by chrismaliszewski** [comment]
Thank you, @BenRQ, for checking that. I didn't have time nor strength; it was almost midnight my time. 
So that makes sense now and explains everything. 

There is only one solution to it which is a good one, especially for a user. 
Right now, as I said before if a creator selects MOM1ET as a required expansion you only see a key icon which tells you nothing and you may think 'ok, I have MOM1EM so I can play' but you can't as the scenario won't work with the expansion selected. 

Solution: we are then coming back to my previous proposition of the hover effect with information on hovering an icon 'what kind of expansions from the type are used/what kind of expansion from the type you're missing' helping a user. 
By doing so, it also helps us as we can get rid of displaying all expansions selected by a creator and instead display only max of these 5 icons when necessary and gain needed space for the proposition
https://user-images.githubusercontent.com/7532602/53086381-ddc4a100-3569-11e9-9c86-9cad00489222.png

**Comment by chrismaliszewski** [comment]
I can imagine the enhancement requires a lot of work and is not something that will be commonly used (rather ones in a while when you buy a new extension). So it can put this at the end of the todo list but I think it'd be better to develop the hover effect (https://github.com/NPBruce/valkyrie/issues/1073) to at least display a list of expansions associated with the icon because, as shown above, it may be problematic to distinguish an expansions.

**Comment by BenRQ** [comment]
I agree hover effect would be quite useful.
